docs/docs/kds-configuration.md

```diff
@@ -54,147 +46,102 @@ enclaveHost.start(null, null, null, kdsConfiguration) {}
+- `useOwnCodeSignerAndProductID` - Use the code signer and product ID of this enclave. This will add the code signer and product ID of the enclave to the `constraint` property at runtime.
+
+### Choosing a value for the `kdsEnclaveConstraint`
+The KDS enclave constraint controls how your enclave will attest to the KDS enclave at the URL provided on startup (see [enclave constraints](constraints.md)).
+
+!!!note
+For the R3 KDS, the following constraint should be used: `S:B4CDF6F4FA5B484FCA82292CE340FF305AA294F19382178BEA759E30E7DCFE2D PROD:1 SEC:STALE`.
+
+### Choosing a value for the `masterKeyType`
+The KDS contains several internal master keys which may be used when deriving keys for user enclaves.
+The master key is an internal secret of the KDS instance that is used during key derivation to prevent other KDS instances from deriving the same keys.
+There are several supported key types, each with different trust models:
+
+- `development` - A stable master key which is not suitable for production workloads, but can be used to test KDS integration. Release mode enclaves cannot use this key type.
+- `akv_hsm` - A production ready master key which is backed by an Azure Key Vault. The key is end-to-end encrypted, however users of this key are assuming trust in Microsoft and R3.
+- `cluster` - A production ready master key which resides entirely within SGX enclaves. Neither R3 nor the KDS hosting provider have access to this key.
+
+### Defining the `policyConstraint` section
+!!!warning
+Selection of the KDS policy constraint can have significant security and operational implications.
+For this reason, users are strongly encouraged to read the following sections in their entirety.
+
+The `policyConstraint` section defines which enclaves can access a derived key.
+It is important to choose a constraint which is strict enough to prevent unauthorised access the key, but also lenient enough to allow for upgrades of enclave code and data migration if required.
+Example configurations are included later in the section.
+
+### Setting the `constraint` property
+The `constraint` property of `policyConstraint` functions in a similar manner to the [`EnclaveConstraint`](api/-conclave/com.r3.conclave.common/-enclave-constraint/index.html) object which is used when attesting to a Conclave enclave.
+When a request for a key is received by the KDS, the KDS attests that the key policy constraint matches the details of the requesting enclave before releasing a derived key.
+Just like Conclave attestation, it follows the [enclave constraints DSL](constraints.md).
+
+### Setting the `useOwnCodeHash` and `useOwnCodeSignerAndProductID` properties
+In addition to the `constraint` property, the `policyConstraint` section has two properties which allow you to include the code hash or code signer and product ID of the current enclave to the constraint.
```

Comment:
The `useOwnCodeHash` and `useOwnCodeSignerAndProductID` properties of the `policyConstraint` section allow you to include the code hash or code signer and product ID of the current enclave to the constraint.

Reply:
I have changed the wording here, but not in exactly the way requested.